protocols/v2/roles-logic-sv2/src/handlers/job_declaration.rs

```diff
@@ -184,5 +182,8 @@ where
         &mut self,
         message: ProvideMissingTransactionsSuccess,
     ) -> Result<SendTo, Error>;
-    fn handle_submit_solution(&mut self, message: SubmitSolutionJd) -> Result<SendTo, Error>;
+    fn handle_submit_solution(
+        self_: Arc<Mutex<Self>>,
```

Comment:
why you change self_ from &mut self_ to Arc<Mutex<Self>> ??

Reply:
because I made a mistake